bff/pkg/server/promql_codegen_test.go

```diff
@@ -59,15 +59,54 @@ func TestGenerateActivityCubePromQLQuery(t *testing.T) {
 	// Track whether any changes were made
 	anyMismatch := false
 
+	now := time.Now()
+
 	for _, tt := range testCases {
 		t.Run(tt.Name, func(t *testing.T) {
-			var input server.PromQLInput
-			err := json.Unmarshal(tt.Input, &input)
+			var rawInput map[string]interface{}
+			err := json.Unmarshal(tt.Input, &rawInput)
 			if err != nil {
 				t.Fatalf("Failed to unmarshal input: %v", err)
 			}
 
-			query, err := server.GenerateActivityCubePromQLQuery(input)
+			startStr, ok := rawInput["start"].(string)
+			if !ok {
+				t.Fatalf("Invalid type for start: %v", rawInput["start"])
+			}
+			endStr, ok := rawInput["end"].(string)
+			if !ok {
+				t.Fatalf("Invalid type for end: %v", rawInput["end"])
+			}
+
+			startTime, err := parseTimeParameter(startStr, now)
+			if err != nil {
+				if !tt.HasError {
+					t.Errorf("unexpected error parsing start time %v", err)
+				}
+				return
+			}
+
+			endTime, err := parseTimeParameter(endStr, now)
```

Comment:
If `endTime` is not specified, it's `now` by default.

Reply:
At the api level, end is a required param, so it is always passed in. i think we should remove that logic from the promql gen, just as we removed the other validation logic. What do you think

Reply:
If that's required, then it's fine and we can remove the logic for setting a default for it.